lib/rules/prefer-destructuring.js

```diff
@@ -82,14 +84,39 @@ module.exports = {
         }
 
         /**
-         * Report that the given node should use destructuring
+         * Find the length to slice the nodeText and the accessor object since destructuring
+         * removes the accessor syntax from the right hand side of the AST
          *
-         * @param {ASTNode} reportNode the node to report
-         * @param {string} type the type of destructuring that should have been done
-         * @returns {void}
+         * @param {Token} lastToken the last token in the reportNode
+         * @param {SourceCode} sourceCode the source code object for simplicity
+         * @returns {Object} removalState the state from the right hand side to move/remove the text
          */
-        function report(reportNode, type) {
-            context.report({ node: reportNode, message: `Use ${type} destructuring` });
+        function findRightHandRemovalState(lastToken, sourceCode) {
```

Comment:
It doesn't really matter either way, but it might be easier to just define `sourceCode` in a closure outside this function rather than passing it in as an argument.